### PR TITLE
Movement API

### PR DESCRIPTION
**Pull request by liran-funaro**

### CODE REVIEW

File: pkg/types/http.go [hunk 1]
```diff
@@ -237,6 +237,20 @@ type FungibleAccountRecord struct {
 	Comment string `json:"comment"`
 }
 
+type FungibleMovementRecord struct {
+	SourceAccounts     []string `json:"sourceAccounts"`     // the source accounts (there may be multiple sources for consolidate)
+	DestinationAccount string   `json:"destinationAccount"` // the receiver account
+	MainBalance        uint64   `json:"mainBalance"`        // the balance of the main account
+	IncomeBalance      uint64   `json:"incomeBalance"`      // the sum of all incoming accounts balance
+}
+
```
Comment (tock-ibm) [inline pkg/types/http.go:246]: I think we need to add some concept or time or order. Surely, the version of the main account, this provides natural "blockchain" time stamp. Once we had the mapping between blockchain version to wall-clock time, we can add that as well.

Maybe even provide versions of all other accounts as well, i.e. have {string, Version} describe every account, and select the appropriate Version of the main account (which should be the highest) as the "movement-timestamp"?

Is `IncomeBalance` the right name? Maybe `IncomingBalance` ?

File: pkg/types/http.go [hunk 2]
```diff
@@ -237,6 +239,30 @@ type FungibleAccountRecord struct {
...
+	MovementVersion *oriontypes.Version `json:"movementVersion"` // the incoming account version for this movement
+	Account         string              `json:"account"`         // the incoming account ID
+	Quantity        uint64              `json:"quantity"`        // the incoming account balance
+	Comment         string              `json:"comment"`         // the incoming account comment
+	SourceOwner     string              `json:"sourceOwner"`     // the user ID that transferred the fungible tokens
+}
+
+type FungibleMovementRecord struct {
+	MovementVersion    *oriontypes.Version              `json:"movementVersion"`              // the main account version for this movement
+	SourceMovements    []FungibleIncomingMovementRecord `json:"sourceMovements,omitempty"`    // the incoming accounts' movements (for consolidate, otherwise empty)
```
[the remark this answers is not in the document]
Comment (liran-funaro) [reply pkg/types/http.go:252]: OK

File: pkg/types/http.go [hunk 3]
```diff
@@ -237,6 +239,30 @@ type FungibleAccountRecord struct {
...
 
+type FungibleIncomingMovementRecord struct {
+	MovementVersion *oriontypes.Version `json:"movementVersion"` // the incoming account version for this movement
+	Account         string              `json:"account"`         // the incoming account ID
+	Quantity        uint64              `json:"quantity"`        // the incoming account balance
+	Comment         string              `json:"comment"`         // the incoming account comment
+	SourceOwner     string              `json:"sourceOwner"`     // the user ID that transferred the fungible tokens
+}
+
+type FungibleMovementRecord struct {
```
[the remark this answers is not in the document]
Comment (liran-funaro) [reply pkg/types/http.go:250]: I will add an explicit "string" field ("outgoing transfer" or "consolidation"). OK?

File: pkg/types/http.go [hunk 4]
```diff
@@ -237,6 +239,30 @@ type FungibleAccountRecord struct {
 	Comment string `json:"comment"`
 }
 
+type FungibleIncomingMovementRecord struct {
+	MovementVersion *oriontypes.Version `json:"movementVersion"` // the incoming account version for this movement
```
[the remark this answers is not in the document]
Comment (liran-funaro) [reply pkg/types/http.go:243]: I think we can put a 128-bit number represented as a HEX string here. But I'm not sure how to name it.

File: pkg/types/http.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tock-ibm) [reply pkg/types/http.go:250]: yes

**Pull request merged**